Get config from server

Nokitoo opened the pull request:
Changed Mage SDK setup process:
- No longer need to feed Webpack with mage.getClientConfig
- Add Mage.setEndpoint to setup server endpoint
- add Mage.init to fetch config from the Mage server

Fixes mage/mage#239

Nokitoo committed:
Get config from server

Changed Mage SDK setup process:
- No longer need to feed Webpack with mage.getClientConfig
- Add Mage.setEndpoint to setup server endpoint
- add Mage.init to fetch config from the Mage server

Fixes mage/mage#239

Nokitoo committed:
Set baseUrl server side

stelcheck commented on docs/ReadMe.md:
```diff
@@ -1,21 +1,21 @@
-examples for specific build systems.
+## Exposing the client config
+
+To init the sdk, your MAGE server needs to expose the client config.
```
SDK, not sdk.

stelcheck commented on docs/ReadMe.md:
```diff
@@ -47,26 +47,47 @@ up yet.
-// register and set up all modules that don't require login
+mage.setEndpoint(
+	'http://127.0.0.1', // Base url
+	'game' 				// App name
```
We could make this optional (default: 'game')

stelcheck commented on docs/ReadMe.md:
```diff
@@ -47,26 +47,47 @@ up yet.
+	'game' 				// App name
+);
+
+// Init mage sdk
```
SDK

stelcheck commented on docs/ReadMe.md:
```diff
@@ -47,26 +47,47 @@ up yet.
+
+mage.init(function(err) {
+	if (err) {
+		return;
```
Should at least log

stelcheck commented on lib/Mage.js:
```diff
@@ -3,22 +3,67 @@ var inherits = require('inherits');
+
+Mage.prototype.init = function (cb) {
+	if (!this.endpoint) {
+		throw new Error('Please call Mage.setEndpoint before calling Mage.init');
```
mage.setInstance; `Mage...` looks like a static method.

stelcheck commented on lib/Mage.js:
```diff
@@ -3,22 +3,67 @@ var inherits = require('inherits');
+	}
+
+	var that = this;
+	var hr = new HttpRequest({
```
I think we discussed this, but can't we use `sendCommand` somehow? For the record, why couldn't we use it?

Nokitoo replied:
Because the `CommandCenter` need the config to setup.

stelcheck reviewed (changes requested):
Overall it goes in the right direction.

Nokitoo committed:
Modify modules loading

- Modules are now loaded automatically and can be setup using a new method returning a Promise
- Removed Webpack integration doc
- Updated ReadMe

Nokitoo committed:
Modify CommandCenter

Commands return promise if no callback passed + can batch commands explicitly

Nokitoo commented:
I don't know if it should be in this PR or a new one, but I think we should definitely remove all client related config from the server config.
For example, `server.msgStream.detect` is only used by the SDK to know in what order it should attempt to create protocols.

stelcheck commented:
We can do this in another PR, yes.

Nokitoo committed:
Add repository in package.json

Closes #6

Nokitoo committed:
Mage.configure better error

- Add Error function thrown in Mage.configure
- Log error in README example

Nokitoo committed:
Fix modules setup

Commands were not added correctly in the module with mage.setupModule. Thus, mage-sdk-js.logger could not find logger.sync even if the MAGE server expose it.

Nokitoo committed:
Modify mage.batchCommands

batchCommands returns an array containing data and error keys for each element.

Nokitoo committed:
Fix msgServer does not start

Nokitoo committed:
Fix typo
